src/Symfony/Component/HttpKernel/ControllerMetadata/Argument/ArgumentMetadataFactory.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\ControllerMetadata\Argument;
+
+/**
+ * Builds method argument data.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class ArgumentMetadataFactory implements ArgumentMetadataFactoryInterface
```

Comment:
In #17933 I have played around with caching controller information and seeing the results, it's worth caching as long as the caching mechanism is fast. Note that I do have an SSD in the system I'm testing on. I have no idea if Reflection is still slower without xdebug, but I'm always happy if I can replace reflection.

With that in mind, that's exactly why I have an interface here.